src/tls.c

```diff
@@ -4420,66 +4420,28 @@ static int TLSX_SetSupportedVersions(TLSX** extensions, const void* data,
 /* Sugnature Algorithms                                                       */
 /******************************************************************************/
 
-#ifdef WOLFSSL_TLS13
 /* Return the size of the SignatureAlgorithms extension's data.
  *
  * data  Unused
  * returns the length of data that will be in the extension.
  */
-static word16 TLSX_SignatureAlgorithms_GetSize(byte* data)
+static word16 TLSX_SignatureAlgorithms_GetSize(void* data)
 {
     WOLFSSL* ssl = (WOLFSSL*)data;
-    int      cnt = 0;
-
-    (void)data;
 
-#ifndef NO_RSA
-    #ifndef NO_SHA1
-        cnt++;
-    #endif
-    #ifndef NO_SHA256
-        cnt++;
-    #endif
-    #ifdef HAVE_SHA384
-        cnt++;
-    #endif
-    #ifdef HAVE_SHA512
-        cnt++;
-    #endif
-    #ifdef WC_RSA_PSS
-        if (IsAtLeastTLSv1_3(ssl->version)) {
-        #ifndef NO_SHA256
-            cnt++;
-        #endif
-        #ifdef HAVE_SHA384
-            cnt++;
-        #endif
-        #ifdef HAVE_SHA512
-            cnt++;
-        #endif
-        }
-    #endif
-#endif
+    return OPAQUE16_LEN + ssl->suites->hashSigAlgoSz;
+}
 
-#ifdef HAVE_ECC
-    #if !defined(NO_ECC256)  || defined(HAVE_ALL_CURVES)
-        #ifndef NO_ECC_SECP
-            cnt++;
-        #endif
-    #endif
-    #if !defined(NO_ECC384)  || defined(HAVE_ALL_CURVES)
-        #ifndef NO_ECC_SECP
-            cnt++;
-        #endif
-    #endif
-    #if !defined(NO_ECC521)  || defined(HAVE_ALL_CURVES)
-        #ifndef NO_ECC_SECP
-            cnt++;
-        #endif
-    #endif
-#endif
+static void TLSX_SignatureAlgorithms_MapPss(WOLFSSL *ssl, byte* input,
+                                            word16 length)
+{
+    word16 i;
 
-    return OPAQUE16_LEN + cnt * OPAQUE16_LEN;
+    ssl->pssAlgo = 0;
+    for (i = 0; i < length; i += 2) {
+        if (input[i] == rsa_pss_sa_algo && input[i + 1] <= sha512_mac)
```

Comment:
We can't rely on side effects that may change in the future that aren't documented and easily enforced.  If we're going to access i+1 in a loop we should validate i+1 is legal to access.